proposed/disable-multi-level-lookup-by-default.md

```diff
@@ -115,20 +129,24 @@ this is yet to be determined.
 
 ### Way to dismiss informational message about the change
 
-There should be some way to dismiss the information messages printed by
-CLI commands if they are affected.
+This is mainly applicable to CLI commands used on a project with `global.json`
+specifying an SDK version. These would produce a message on every execution
+per the above rules. The `dotnet --info` and error cases are not problematic
+and can show the message always.
+
+Options:
 
+1. Only show the message in problematic places (like CLI commands) during
+   .NET 7 previews, but remove them for final release. 
 1. Require the user to set `DOTNET_MULTILEVEL_LOOKUP=0` - which will disable
-   the message.
+   the message. Would go against our plans to completely remove the feature
+   in the future as we would train users to set this anyway.
 1. Only show the message once (would require a way to remember that it was shown,
-   similar to first run CLI experience).
+   similar to first run CLI experience). The timing is problematic, the first
+   invocation could be from a script and thus invisible to the user.
+   Alternative is to show it for a time period after install.
```

Comment:
Something else to consider is a way to suppress the warnings that writes into a file that already gets loaded.  That way, the cost of loading a new file isn't added into the mix, but the user has a way to make them disappear once they no longer want to see them.

Reply:
I think we could try to do a good job at showing the message only when it actually matters. In the error case, we could even go and do the lookup "with MLL enabled" and compare the results, and only show the message if they differ. Not saying we need to do that, but it's possible.

For the SDK lookup part - I wonder if we should add a message always, or just when it fails. The message should be shown only when using private runtime anyway, which is already a relatively small subset of overall uses, and I want to see how much more we can reduce it without taking a noticable perf hit (running the result "with" MLL might be too expensive, but could measure that).

Reply:
Sounds good.  I agree that running "with" is likely too expensive.  I remember digging into the cost of MLL at one point, and thinking that it would not be ideal to have to run it to decide whether or not to expose error messages.  Hopefully, we can get to a point where we only run MLL when explicitly requested for those versions of .NET that support require it.